core/modules/installer/installer.manager.inc

```diff
@@ -424,7 +431,8 @@ function installer_manager_update_ready_form($form, &$form_state) {
  */
 function installer_manager_update_ready_form_submit($form, &$form_state) {
   // Store maintenance_mode setting so we can restore it when done.
-  $_SESSION['maintenance_mode'] = config_get('system.core', 'maintenance_mode');
+  $_SESSION['maintenance_mode'] = state_get('system.core', 'maintenance_mode');
```

Comment:
@jenlampton I think I have faced similar problems when working on https://github.com/backdrop/backdrop-issues/issues/2583 (sorry, it's been almost a year, so can't recall what exactly). Can you please try something like this:

```suggestion
  $previous_maintenance_mode_state = state_get('maintenance_mode', FALSE);
  $maintenance_mode = $form_state['values']['maintenance_mode'];
  state_set('maintenance_mode', (bool) $maintenance_mode);
  // Also store maintenance_mode setting so we can restore it when done.
  $_SESSION['maintenance_mode'] = $maintenance_mode;
```

Notice how the proper way to use `state_get()` is `state_get('maintenance_mode', FALSE);`.

Also noting that in one of my PRs I have this:

```php
  // In some cases, the site might seem as still online/offline, due to cache.
  cache_clear_all();
```

Reply:
outdated... but I added a cache_clear_all() just in case.